### PR TITLE
feat(boundary): provider, injector and component

### PR DESCRIPTION
**Pull request by miguelramos**
Initial propose.

### CODE REVIEW

**Review by tannerlinsley — APPROVED** [review]

**Review by Shyrro — COMMENTED** [review]
I didn't test it yet but took a look at the code and looks fine to me.

File: example/app/components/HelloWorld.vue [hunk 1]
```diff
@@ -23,9 +28,14 @@ export default {
   },
   setup() {
     const client = useQueryClient()
+    const queryError = useQueryErrorResetBoundary()
```
Comment (miguelramos) [inline example/app/components/HelloWorld.vue:31]: I need to test where if the use of a destruct will make lose reference. An it yes i need to put toRefs on BoundaryProvider.

File: src/vue/QueryErrorResetBoundary.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+export const useQueryErrorResetBoundary = () => inject(QueryErrorResetBoundarySymbol);
+
+export const useQueryErrorResetBoundaryProvider = () => {
+  const isResetRef = ref(false);
+
+  const api = {
+    clearReset: () => { isResetRef.value = false; },
+    reset: () => { isResetRef.value = true; },
+    isReset: () => isResetRef.value
```
Comment (iosamuel) [inline src/vue/QueryErrorResetBoundary.ts:19]: We could just make it the isReseted computed value and export it

**Review by iosamuel — COMMENTED** [review]
isReseted computed as part of the API

File: src/vue/QueryErrorResetBoundary.ts (hunk 2 above)
Comment (miguelramos) [reply src/vue/QueryErrorResetBoundary.ts:19]: Do you mean isReset? Typo? I would go for computed if it was a property but the is reset function will always get the native value setted by the other two functions.

Comment (miguelramos) [reply src/vue/QueryErrorResetBoundary.ts:19]: Ah ok! What you saw in the example hello world put it integrated on the boundary.. Yep cool. Sorry didn't catch on first comment.

File: example/app/components/HelloWorld.vue (hunk 1 above)
Comment (miguelramos) [reply example/app/components/HelloWorld.vue]: Tested and working

File: src/vue/QueryErrorResetBoundary.ts (hunk 2 above)
Comment (miguelramos) [reply src/vue/QueryErrorResetBoundary.ts]: Implemented

**Pull request merged**